File: SocialPulse/UI/mobile_socialpulse/lib/models/message.dart

import 'package:json_annotation/json_annotation.dart';
import 'package:mobile_socialpulse/models/image.dart';

part 'message.g.dart';

@JsonSerializable()
class Message{
  int? id;
  String? text;
  int? userId;
  int? conversationId;
  DateTime? createdAt;
  List<Image>? images;

  Message (this.id,this.text,this.userId,this.conversationId, this.createdAt, this.images);

  factory Message.fromJson(Map<String, dynamic> json) =>
      _$MessageFromJson(json);

  Map<String, dynamic> toJson() => _$MessageToJson(this);
}

File: SocialPulse/SocialPulse.Core/Entities/Message.cs
namespace SocialPulse.Core
{
    public class Message : BaseEntity
    {
        public string Text { get; set; } = null!;

        public int UserId { get; set; }
        public User User { get; set; } = null!;
        public int ConversationId { get; set; }
        public Conversation Conversation { get; set; } = null!;
        public ICollection<Image> Images { get; set; } = null!;

    }
}
